### DISCUSSION TITLE
How to get the text data from the current tab by using some code.

### DISCUSSION THREAD

**Discussion by wochatme** [microsoft/terminal#17918] [General]
I am adding some code in NonClientIslandWindow.cpp to response to some windows event such as WM_LBUTTONUP. I want to get the text data from the current terminal window, like a LPCWSTR pointer.  I checked the code, and some some useful functions like:

```
        const auto visible = _GetVisibleViewport();
        const auto top = visible.Top();
        const auto height = visible.Height();

Terminal::TextCopyData Terminal::RetrieveSelectedTextFromBuffer
```

Now I am blocked how to call these functions from a function responding WM_LBUTTONUP event of NonClientIslandWindow window.

Could someone give me some idea?

Much appreciated!

**Comment by zadjii-msft** [comment]
Sorry, what exactly are you trying to accomplish, big-picture? 

The answer here probably changes if you're building something inside the Terminal codebase, vs in another process that's trying to hook the Terminal's wndproc 😉

**Reply by wochatme** [reply]
I add a patch into the code of NonClientIslandWindow.cpp. My requirement is: if the event of WM_LBUTTONUP is triggered, one function is called. Within this function, I want to capture the text data of the current terminal window as LPCWSTR or std::wstring.
